### ISSUE TITLE
Allow more input types for application description

### ISSUE THREAD

**Issue by rivernews** [rivernews/appl-tracky-spa#21]
### Current Behavior
- Application has an input for a url link to the job description page.
### Desired Behavior
- Will be nice to have a textfield to have a short note for the application.
- Also one user pointed out company may pull down the job description page once the interview starts. In such case there's no way to retrieve the description by url or online. 
    - One suggestion the user gave is to allow screenshot image upload.

To sum up:
- Image upload field
- Textfield

**Comment by rivernews** [comment]
### Scalability for text field input

Ideally, the text field can contain rich content. If so, we can add link or even image then the image upload field is not needed. 

But the application status link becomes less useful too. However, the status text can come in handy serving for tab labels. [See issue for tab](https://github.com/rivernews/appl-tracky-spa/issues/6).

**Comment by rivernews** [comment]
### Implementation for workaround

- [ ] Add the new field in backend model, for all company, application and application status.
- [ ] Add new fields to serializer
- [ ] Add new fields in frontend data models
- [ ] Add new fields in frontend forms
- [ ] Add new fields in frontend render components

**Comment by rivernews** [comment]
### Working with tab

One way to do it is
- Make the current status text be used as simple text field.
- At the company model level, add a status field for tab.
    - Whether we use the current many-to-many label field, or we use a new choice field, is however arguable.

**Comment by rivernews** [comment]
Update:

- Status: may related to tabbed UI 
- Rich text editor for application, or for company

**Comment by rivernews** [comment]
### Add Rich Text Field for Application

Aspects
- [ ] Backend model
- [ ] Backend CRUD consideration?
- [ ] Frontend data model update?
- [ ] Frontend editor onkeyboardDown - update the data mechanism?
